### ISSUE TITLE
Build error on Ubuntu 22.04

### ISSUE THREAD

**Issue by BLCRAFT210** [Felk/dolphin#29]
After pulling 26cdedc, I have a new build error. I have attached stderr below.
[log.txt](https://github.com/Felk/dolphin/files/10346579/log.txt)

**Comment by Felk** [comment]
That seems to be an issue with GCC 11. I tried compiling with GCC 12 but that just causes a different slew of issues :/
For reference what, I tried:
```shell
rm -rf ./build/
sudo apt install gcc-12 g++-12
mkdir build && cd build
CC=/usr/bin/gcc-12 CXX=/usr/bin/g++-12 cmake ..
make -j $(nproc)
```
[dolphin-gcc12-error.txt](https://github.com/Felk/dolphin/files/10352228/dolphin-gcc12-error.txt)

I'll try again at a later time

**Comment by BLCRAFT210** [comment]
Do you know when you can resolve this build error? Currently I'm working off of c48006d and the memory leak will be a problem for me sometime soon.

**Comment by Felk** [comment]
I made a bit of progress today and will try to get some more work in this week, but I probably won't be able to fix this issue on a short notice, sorry

**Comment by BLCRAFT210** [comment]
Any more progress on this?

**Comment by Felk** [comment]
Nope, I'm sorry. Not because the problem is especially hard, but because I just haven't opened Visual Studio in months 😅 Thanks for asking though

**Comment by BLCRAFT210** [comment]
I spent a while with my friend trying to fix this. It looks like all we needed to do was to build with clang 😂

I'll make sure tomorrow that the build is functional!

**Comment by Felk** [comment]
I just tried the newest master with gcc12 and it seems to work. If I find a source that upstream also lifted their minimum gcc version to 12 I'll close this issue, but they still explicitly reference gcc11 in their wiki, so probably not...

So workarounds right now are: Use GCC 12 or Clang on Ubuntu. I'll leave this issue open for now
